src/dlmalloc.rs

```diff
@@ -24,31 +25,9 @@ pub struct Dlmalloc {
     trim_check: usize,
     least_addr: *mut u8,
     release_checks: usize,
+    _system_allocator: PhantomData<S>,
```

Comment:
Some platforms may need to store data so could this be `S` instead of `PhantomData`?

Reply:
Sure thing. Well, I changed it to `Option<S>` so `pub const fn init() -> Dlmalloc<Sys>` could be implemented.